sdk/identity/azure-identity/migration_guide.md

```diff
@@ -0,0 +1,38 @@
+# Guide for migrating to azure-identity from azure-common
+
+## JSON- and file-based authentication
+
+To encourage best security practices, `azure-identity` does not support JSON- and file-based authentication in the same
+way as `azure-common`. `azure-common` provided factory methods like [`get_client_from_json_dict`][client_from_json] and
+[`get_client_from_auth_file`][client_from_auth_file] to provide credentials as plaintext JSON. Azure credentials, like
+passwords, should not be stored in plaintext whenever possible. You can find information about Azure Active Directory
+(Azure AD) and best practices for authentication in [Azure AD's documentation][azure_ad].
```

Comment:
I think you can drop the last two sentences. Boils it down to "to encourage more secure practices, we didn't carry forward these methods", and the link is a rabbit hole.